Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -680,6 +611,98 @@ public class PBXProjGenerator {
+                if searchForPlist {
+                    plistPath = getInfoPlist(target.sources)
+                    searchForPlist = false
+                }
+                if let plistPath = plistPath {
+                    buildSettings["INFOPLIST_FILE"] = plistPath.byRemovingBase(path: project.basePath)
+                }
+            }
+            
+            // automatically calculate bundle id
+            if let bundleIdPrefix = project.options.bundleIdPrefix,
+                !project.targetHasBuildSetting("PRODUCT_BUNDLE_IDENTIFIER", basePath: project.basePath, target: target, config: config) {
+                let characterSet = CharacterSet.alphanumerics.union(CharacterSet(charactersIn: "-.")).inverted
+                let escapedTargetName = target.name
+                    .replacingOccurrences(of: "_", with: "-")
+                    .components(separatedBy: characterSet)
+                    .joined(separator: "")
+                buildSettings["PRODUCT_BUNDLE_IDENTIFIER"] = bundleIdPrefix + "." + escapedTargetName
+            }
+            
+            // automatically set test target name
+            if target.type == .uiTestBundle,
+                !project.targetHasBuildSetting("TEST_TARGET_NAME", basePath: project.basePath, target: target, config: config) {
+                for dependency in target.dependencies {
+                    if dependency.type == .target,
+                        let dependencyTarget = project.getTarget(dependency.reference),
+                        dependencyTarget.type == .application {
+                        buildSettings["TEST_TARGET_NAME"] = dependencyTarget.name
+                        break
+                    }
+                }
+            }
+            
+            // objc linkage
+            let anyDependencyRequiresObjCLinking = targetDependencies.contains { dependency in
+                guard dependency.type == .target else { return false }
+                guard let dependencyTarget = project.getTarget(dependency.reference) else { return false }
+                guard targetsLinked.contains(dependencyTarget.name) else { return false }
+                
+                func dependencySourceFiles() -> [SourceFile] {
+                    // This is duplicative and should be cached
+                    return (try? sourceGenerator.getAllSourceFiles(sources: dependencyTarget.sources)) ?? []
+                }
+                
+                return dependencyTarget.requiresObjCLinking
+                    ?? dependencySourceFiles().contains { ["m", "mm", "swift"].contains($0.path.extension) }
+            }
+            if anyDependencyRequiresObjCLinking {
+                let otherLinkingFlags = "OTHER_LDFLAGS"
+                if let string = buildSettings[otherLinkingFlags] as? String {
+                    buildSettings["OTHER_LDFLAGS"] = "\(string) -ObjC"
+                } else {
+                    buildSettings["OTHER_LDFLAGS"] = "$(inherited) -ObjC"
```

Comment:
This is the new code in here. Lots of red and green since this section had to be moved below `targetDependencies` generation